### PR TITLE
Move into a soft-exit in case of kubelet error

### PR DESCRIPTION
**Pull request by mangelajo**
Depends-On: #575

Signed-off-by: Miguel Angel Ajo <majopela@redhat.com>

### CODE REVIEW

**Comment by mangelajo** [comment]
Here I'm changing a little bit the behavior, so a kustomization failure becomes fatal. This may be up for discussion.

**Comment by mangelajo** [comment]
@fzdarsky @oglok @copejon @sseif-rh  I rebased this, please let's review/merge.

File: pkg/node/kubelet.go [hunk 1]
```diff
@@ -178,10 +178,13 @@ func (s *KubeletServer) Run(ctx context.Context, ready chan<- struct{}, stopped
 
 	kubeletDeps, err := kubelet.UnsecuredDependencies(kubeletServer, utilfeature.DefaultFeatureGate)
 	if err != nil {
-		klog.Fatalf("Error in fetching depenedencies", err)
+		klog.Errorf("Error in fetching kubelet depenedencies", err)
```
Comment (oglok) [inline pkg/node/kubelet.go:181]: There is a typo in "dependencies"

**Review by oglok — COMMENTED** [review]
The rest lgtm

**Comment by mangelajo** [comment]
ciosing this one, as this is covered by the global panic capture

**Pull request closed without merging**